pysindy/optimizers/constrained_sr3.py

```diff
@@ -362,7 +364,7 @@ def _objective(self, x, y, q, coef_full, coef_sparse, trimming_array=None):
             R2 *= trimming_array.reshape(x.shape[0], 1)
 
         if self.thresholds is None:
-            regularization = self.reg(coef_full, self.threshold**2 / self.nu)
+            regularization = super().reg(coef_full, self.threshold**2 / self.nu)
```

Comment:
Only call `super()` if this class overrides `SR3.reg`.

Imagine if a subclass (e.g. `TrappingSR3`) later overrides `reg()` but not `_update_coef_cvxpy()`  Then, when that class calls `self._update_coef_cvxpy()`, it may expect `_update_coef_cvxpy()` to refer to its own `TrappingSR3.reg()`.  This is the case if `ConstrainedSR3._update_coef_cvxpy()` uses `self.reg()`.  however, if it uses `super().reg()`, it will always act as `SR3.reg()`